src/components/Autocompleted.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+
+interface Props {
+  people: Person[],
+  timer?: number,
+  onSelect?: (person: Person | null) => void,
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  timer = 1000,
+  onSelect = () => { },
+}) => {
+  const [isFocused, setIsFocused] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, timer),
+    [timer],
+  );
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(p => {
+      return p.name.toLowerCase().includes(appliedQuery.toLowerCase());
+    });
+  }, [people, appliedQuery]);
+
+  const handlePersonSelect = (person: Person | null) => {
+    onSelect(person);
+
+    if (person) {
+      setQuery(person.name);
+    }
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value;
+
+    setQuery(inputValue);
+    applyQuery(inputValue);
+
+    if (inputValue.length === 0) {
+      handlePersonSelect(null);
+    }
+  };
+
+  return (
+    <div className={classNames('dropdown', {
+      'is-active': isFocused,
+    })}
+    >
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onFocus={() => setIsFocused(true)}
+          onBlur={() => setIsFocused(false)}
```

Comment:
It's a bad practice to create anonymous function is jsx, it's better to create separate function for that

Reply:
Check other places